internal/graph/check.go

```diff
@@ -835,11 +785,10 @@ func checkIntersectionTupleToUserset(
 			}
 
 			// Add any caveats on the subject from the starting relationship(s) as well.
-			subjectKey := tuple.StringONR(subject)
-			tuples, _ := relationshipsBySubjectONR.Get(subjectKey)
-			for _, relationTuple := range tuples {
-				if relationTuple.Caveat != nil {
-					caveats = append(caveats, wrapCaveat(relationTuple.Caveat))
+			resourceIDAndCaveats := checksToDispatch.mappingsForSubject(subject.Namespace, subject.ObjectId, subject.Relation)
+			for _, riac := range resourceIDAndCaveats {
+				if riac.caveat != nil {
```

Comment:
`mappingForSubject` asserts that a caveat exists. In runtime it wouldn't panic, and this portion would just ignore the resource, which does not seem right, shouldn't we return a spiceerror?

Reply:
`mappingForSubject` ensures that the *resource* ID exists; not that it is caveated. If there is no caveat, this entry will be `nil`